### PR TITLE
Fix message not send on package API request

### LINKED ISSUE #760: anitya.project.map.new not send when adding new mapping through APIv2
There is missing message publish call when adding new package mapping using APIv2. This causing issue when project was added, new version found, but the mapping is added later.

### PR DESCRIPTION
**Pull request by Zlopez**
Fixes #760 

Signed-off-by: Michal Konečný <mkonecny@redhat.com>

### CODE REVIEW

File: anitya/tests/test_flask_api_v2.py [hunk 1]
```diff
@@ -481,6 +483,22 @@ def test_valid_request(self):
...
+            "project": dict(project.__json__()),
+            "distro": {"name": "Fedora"},
+            "message": {
+                "agent": "user@fedoraproject.org",
+                "project": "requests",
+                "distro": "Fedora",
+                "new": "python-requests",
+            },
+        }
+        mock_publish.assert_called_with(topic="project.map.new", msg=msg)
```
Comment (cverna) [inline anitya/tests/test_flask_api_v2.py]: you use fedora-messaging `testing.mock_sends` for that if you prefer. See example here --> https://pagure.io/waiverdb/blob/master/f/tests/test_events.py#_31

Comment (Zlopez) [reply anitya/tests/test_flask_api_v2.py]: Thanks, I will change it.

Comment (Zlopez) [reply anitya/tests/test_flask_api_v2.py]: I tried to implement the `testing.mock_sends`, but I'm getting this in tox:
```
TypeError: test_valid_request() missing 1 required positional argument: 'mock_publish'
```

The method looks like this now:
```python
    def test_valid_request(self, mock_publish):
        """Assert packages can be created."""
        request_data = {
            "project_ecosystem": "pypi",
            "project_name": "requests",
            "package_name": "python-requests",
            "distribution": "Fedora",
        }
        project = models.Project.query.filter_by(
            name="requests", ecosystem_name="pypi"
        ).one()

        msg = {
            "project": dict(project.__json__()),
            "distro": {"name": "Fedora"},
            "message": {
                "agent": "user@fedoraproject.org",
                "project": "requests",
                "distro": "Fedora",
                "new": "python-requests",
            },
        }

        expected_msg = api.Message(
            topic="anitya.project.map.new",
            body=msg
        )

        with testing.mock_sends(expected_msg):
            output = self.app.post(
                "/api/v2/packages/", headers=self.auth, data=request_data
            )

        self.assertEqual(output.status_code, 201)
        data = _read_json(output)
        self.assertEqual({"distribution": "Fedora", "name": "python-requests"}, data)
```

Comment (Zlopez) [reply anitya/tests/test_flask_api_v2.py]: I will leave it like it is now, because of the change in `project` object that is done inside the post request.

**Pull request merged**